### PR TITLE
[Rules] Add Origin Rules (beta)

### PR DESCRIPTION
**Pull request by pedrosousa**
Addresses PCX-4059.

### CODE REVIEW

File: content/rules/origin-rules/create-api.md [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+    * **kind**: `zone`
+    * **phase**: `http_request_origin`
+
+3. Use the [Update ruleset](/ruleset-engine/rulesets-api/update/) method to add an Origin Rule to the list of ruleset rules (check the examples below). Alternatively, include the rule in the [Create ruleset](/ruleset-engine/rulesets-api/create/) request mentioned in the previous step.
+
+## Required API token permissions
+
+The API token used in API requests to manage Origin Rules must have at least the following permission:
+
+* Origin: Edit
```
Comment (marciocloudflare) [inline content/rules/origin-rules/create-api.md]: Should any of these be bold or in between backticks? I feel the second one might (`Edit`) but I'm not sure

File: content/rules/origin-rules/features.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+# Available features
+
+The following sections describe the features currently supported by Origin Rules.
+
+## Host Header Override
+
+This feature allows you to rewrite the HTTP `Host` header of incoming requests.
+
+A common use case for this functionality is when your content is hosted on an a third-party server that only accepts `Host` headers with their own server names. In this situation, you must update the `Host` HTTP header in incoming requests from `Host: example.com` to `Host: thirdpartyserver.example.net`.
```
Comment (marciocloudflare) [inline content/rules/origin-rules/features.md]:
```suggestion
A common use case for this functionality is when your content is hosted on a third-party server that only accepts `Host` headers with their own server names. In this situation, you must update the `Host` HTTP header in incoming requests from `Host: example.com` to `Host: thirdpartyserver.example.net`.
```

File: content/rules/origin-rules/features.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+* A hostname on the same Cloudflare account (possibly on a different zone).
+* A hostname for which Cloudflare is not proxying traffic (gray-clouded).
+
+## Destination Port Override
+
+This feature allows you to override the destination port of a request.
+
+When you configure Destination Port Override, you can redirect incoming requests to a different port. For example, you could override the destination port for requests received for `mydomain.com` so that they are served by the application running on port 9000 (`mydomain.com:9000`).
+
+The destination port must be between 1 and 65535.
```
Comment (marciocloudflare) [inline content/rules/origin-rules/features.md]:
```suggestion
The destination port must be between 1 and 65,535.
```

Comment (marciocloudflare) [reply content/rules/origin-rules/features.md]: Our style guide mentions we should add commas for numbers with more than 3 digits

File: content/rules/origin-rules/parameters.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+meta:
+  title: Origin Rules API parameter reference
+---
+
+# API parameter reference
+
+Create [different overrides](/rules/origin-rules/features/) by including different action parameters in the `action_parameters` field:
+
+*   Define the `host_header` parameter to override the HTTP `Host` header of incoming requests.
+*   Define the `origin` object to override the hostname and/or destination port of incoming requests.
```
Comment (marciocloudflare) [inline content/rules/origin-rules/parameters.md]:
```suggestion
* Define the `host_header` parameter to override the HTTP `Host` header of incoming requests.
* Define the `origin` object to override the hostname and/or destination port of incoming requests.
```

**Review by marciocloudflare — APPROVED** [review]

File: content/rules/origin-rules/create-api.md (hunk 1 above)
Comment (pedrosousa) [reply content/rules/origin-rules/create-api.md]: Hmm, I guess I can use two backticked regions of text here.  The API will return a different description ("Origin Write"), but the dashboard will have two drop-down lists: one for selecting `Origin`, and a sub-dropdown for selecting `Edit`.

File: content/rules/origin-rules/features.md (hunk 2 above)
Comment (pedrosousa) [reply content/rules/origin-rules/features.md]: Great catch!

File: content/rules/origin-rules/create-api.md (hunk 1 above)
Comment (pedrosousa) [reply content/rules/origin-rules/create-api.md]: Maybe two italicized sections, since it's something the user selects in the interface.

**Pull request merged**